.azure/pipelines/Phonebook.preview.yml

```diff
@@ -0,0 +1,193 @@
+pr: 
+  branches:
+    include:
+      - master
+
+variables:
+  image_repo: danielhabenicht/phonebook
+  image_tag: pr-$(System.PullRequest.PullRequestNumber) 
+  helm_artifact_name: HelmPhonebookPackage
+
+  jobs:
+    - job: build_phonebook_frontend
+      displayName: 'Phonebook Frontend'
+      pool:
+        vmImage: 'ubuntu-16.04'
+      steps:
+      - task: UseNode@1
+        inputs:
+          version: '11.x'
+
+      - script: node version.js $(image_tag) $(Build.SourceVersion) $(Build.SourceVersion)
+        displayName: 'Write Version Number'    
+        workingDirectory: 'Phonebook.Frontend/'
+
+      - script: docker build -t $(image_repo):$(image_tag) .
+        displayName: 'Build Image'              
+        workingDirectory: 'Phonebook.Frontend/'
+      
+      - script: |
+          docker login -u $DOCKER_REGISTRY_USER -p $(DOCKER_REGISTRY_PASSWORD)
+          docker push $(image_repo):$(image_tag)
+        workingDirectory: 'Phonebook.Frontend/'
+        displayName: 'Push Image'                      
+        env:
+          DOCKER_REGISTRY_PASSWORD: $(DOCKER_REGISTRY_PASSWORD)
+          DOCKER_REGISTRY_USER: $(DOCKER_REGISTRY_USER)
+
+    - job: build_phonebook
+      displayName: Phonebook
+      pool:
+        vmImage: 'ubuntu-16.04'
+      steps:
+      - task: HelmInstaller@1
+        inputs:
+          helmVersionToInstall: 'latest'
+      - script: helm dep update
+        workingDirectory: 'Phonebook/phonebook/'
+      - task: HelmDeploy@0
+        displayName: Helm package
+        inputs:
+          command: package
+          chartPath: Phonebook/phonebook
+          destination: $(Build.ArtifactStagingDirectory)
+      - publish: $(Build.ArtifactStagingDirectory)
+        artifact: $(helm_artifact_name)
+        
+    - job: deploy_preview
+      dependsOn:
+       - build_phonebook_frontend
+       - build_phonebook
+      displayName: 'Deploy Preview' 
+      pool:
+        vmImage: 'ubuntu-16.04'
+      variables:
+        namespace: $(image_tag)
+        azure_public_ip_name: namespace_$(namespace)_public_ip
+      steps:
+      - download: current
+        artifact: $(helm_artifact_name)
+
+      - task: HelmInstaller@0
+        displayName: 'Install Helm 2.14.2'
+        inputs:
+          helmVersion: 2.14.2
+      
+      - task: AzureCLI@1
+        displayName: 'Azure CLI '
+        inputs:
+          azureSubscription: 'Visual Studio Enterprise – MPN (44678328-75c5-4e48-92c0-b2ab31d5166c)'
```

Comment:
because of this a build shouldn't deploy... but I understand the mind of transparency etc...

Reply:
this can be solved